Source/Scene/GroundPolylinePrimitive.js

```diff
@@ -0,0 +1,759 @@
+        '../Core/Check',
+        '../Core/ComponentDatatype',
+        '../Core/defaultValue',
+        '../Core/defined',
+        '../Core/defineProperties',
+        '../Core/destroyObject',
+        '../Core/DeveloperError',
+        '../Core/GeometryInstance',
+        '../Core/GeometryInstanceAttribute',
+        '../Core/GroundPolylineGeometry',
+        '../Core/isArray',
+        '../Core/Matrix4',
+        '../Shaders/PolylineShadowVolumeVS',
+        '../Shaders/PolylineShadowVolumeFS',
+        '../Shaders/PolylineShadowVolumeMorphVS',
+        '../Shaders/PolylineShadowVolumeMorphFS',
+        '../Renderer/DrawCommand',
+        '../Renderer/Pass',
+        '../Renderer/RenderState',
+        '../Renderer/ShaderProgram',
+        '../Renderer/ShaderSource',
+        '../ThirdParty/when',
+        './BlendingState',
+        './CullFace',
+        './Material',
+        './PolylineColorAppearance',
+        './PolylineMaterialAppearance',
+        './Primitive',
+        './SceneMode'
+    ], function(
+        ApproximateTerrainHeights,
+        Check,
+        ComponentDatatype,
+        defaultValue,
+        defined,
+        defineProperties,
+        destroyObject,
+        DeveloperError,
+        GeometryInstance,
+        GeometryInstanceAttribute,
+        GroundPolylineGeometry,
+        isArray,
+        Matrix4,
+        PolylineShadowVolumeVS,
+        PolylineShadowVolumeFS,
+        PolylineShadowVolumeMorphVS,
+        PolylineShadowVolumeMorphFS,
+        DrawCommand,
+        Pass,
+        RenderState,
+        ShaderProgram,
+        ShaderSource,
+        when,
+        BlendingState,
+        CullFace,
+        Material,
+        PolylineColorAppearance,
+        PolylineMaterialAppearance,
+        Primitive,
+        SceneMode) {
+    'use strict';
+
+    /**
+     * A GroundPolylinePrimitive represents a polyline draped over the terrain in the {@link Scene}.
+     * <p>
+     *
+     * Only to be used with GeometryInstances containing GroundPolylineGeometries
+     *
+     * @param {Object} [options] Object with the following properties:
+     * @param {Array|GeometryInstance} [options.geometryInstances] GeometryInstances containing GroundPolylineGeometry
+     * @param {Appearance} [options.appearance] The Appearance used to render the polyline. Defaults to a white color {@link Material} on a {@link PolylineMaterialAppearance}.
+     * @param {Boolean} [options.show=true] Determines if this primitive will be shown.
+     * @param {Boolean} [options.vertexCacheOptimize=false] When <code>true</code>, geometry vertices are optimized for the pre and post-vertex-shader caches.
+     * @param {Boolean} [options.interleave=false] When <code>true</code>, geometry vertex attributes are interleaved, which can slightly improve rendering performance but increases load time.
+     * @param {Boolean} [options.compressVertices=true] When <code>true</code>, the geometry vertices are compressed, which will save memory.
```

Comment:
Similar comment - can/should this basically be done inline when the geometry is computed?

Basically the "geometry" and "primitive" are more tightly coupled than our general geometry and primitive types so I suspect there are some reasonable design decisions that could lead to better creation performance and less code.

Reply:
I guess there isn't actually anything in the geometry here that should be compressed, I'll just take this out.